### PR TITLE
chore(examples): migrate to v2 manifest

### PR DESCRIPTION
**Pull request by vdice**
Ref https://github.com/fermyon/spin/issues/1947

I tried to keep all changes to just the `spin.toml` manifest file for each example, but cracked and had to include the following add'l changes as I tested/ran the examples:

- Updates to a few outdated tinygo example READMEs
- Minor update to the spin-timer README
- Fix for the wasi-http-streaming-file to reference the correct file
- Rename `config-tinygo` to `variables-tinygo` in line with pre-existing `variables-rust` and our no longer using "config" in the manifest

### CODE REVIEW

**Comment by lann** [comment]
Sorry, I had meant to get a `spin doctor` migrator in place for this.

**Review by itowlson — APPROVED** [review]
Only suggestion is very non-blocking and since your eyes must be bleeding by now I'm good with picking it up at some much later date!

File: examples/http-rust-outbound-http/spin.toml [hunk 1]
```diff
@@ -1,48 +1,57 @@
...
+route = "/hello"
+component = "hello-component"
+
+[component.outbound-http]
 source = "outbound-http/target/wasm32-wasi/release/http_rust_outbound_http.wasm"
 allowed_http_hosts = ["https://random-data-api.fermyon.app"]
-[component.trigger]
-route = "/outbound"
-[component.build]
+
```
[the remark this answers is not in the document]
Comment (vdice) [reply examples/http-rust-outbound-http/spin.toml:28]: Oh thank you @itowlson -- I meant to ask about preferred style for the manifests.  Indeed, I'd like to align these to represent our suggested formatting.  I see the templates you've updated follow your suggestion above, so I'll do the same.  Let me know if there are any other style suggestions -- might as well implement them in this PR.

File: examples/http-rust-outbound-http/spin.toml [hunk 2]
```diff
@@ -1,48 +1,57 @@
...
+[component.outbound-http-wildcard.build]
 workdir = "outbound-http"
 command = "cargo build --target wasm32-wasi --release"
 
-[[component]]
-id = "outbound-http-to-same-app"
+[component.outbound-http-to-same-app]
 source = "outbound-http-to-same-app/target/wasm32-wasi/release/outbound_http_to_same_app.wasm"
 # To make outbound calls to components in the same Spin app, use the special value self.
 # This is not yet supported in cloud.
```
[the remark this answers is not in the document]
Comment (vdice) [reply examples/http-rust-outbound-http/spin.toml:44]: Indeed, I'm pretty sure it is.  I'll create a separate PR to remove...

File: examples/http-tinygo/README.md [hunk 3]
```diff
@@ -1,75 +1,35 @@
...
-func main() {}
-```
-
-> For more information and examples for using TinyGo with WebAssembly, check
-> [the official TinyGo documentation](https://tinygo.org/docs/guides/webassembly/)
-> and
-> [the Wasm examples](https://github.com/tinygo-org/tinygo/tree/release/src/examples/wasm).
-
-Building this as a WebAssembly module can be done using the `tinygo` compiler:
+# Spin component in TinyGo
```
[the remark this answers is not in the document]
Comment (vdice) [reply examples/http-tinygo/README.md:1]: 😂 good point... let me see about a more specific title.

[the remark this answers is not in the document]
Comment (vdice) [reply examples/http-tinygo/README.md:1]: https://github.com/fermyon/spin/pull/1962/commits/e77580c76c75b2b7df4e8ccbd41189b6c2e65f3e

File: examples/http-rust-outbound-http/spin.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (vdice) [reply examples/http-rust-outbound-http/spin.toml]: https://github.com/fermyon/spin/pull/1962/commits/ffbe4f440e26084e9c947984f1f1bfa55c7042ff

File: examples/http-rust-outbound-http/spin.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (vdice) [reply examples/http-rust-outbound-http/spin.toml:42]: https://github.com/fermyon/spin/pull/1968

**Pull request merged**